### DISCUSSION TITLE
How to implement one-click sign in in Figma plugin?

### DISCUSSION THREAD

**Discussion by hallee9000** [supabase/supabase#6322] [Questions] [answered]
Hey guys, I have a question recently. I'm creating my Figma plugin and I want to implement sign-in in it. But I can't use the magic link in Figma plugin because it won't redirect to the plugin after verification. Also, I don't want to use the email/password method because it's inconvenient for users inputting so much information.

I want to implement a user flow like the below:

- Users click the button in the plugin and it'll open a web page in the browser.
- Users complete sign-in on the web page.
- The plugin will get auth state from supabase server, this user signed in

<img width="814" alt="image" src="https://user-images.githubusercontent.com/12868055/161929381-74813f20-ce9f-477a-a8a9-d755f3b0e8a9.png">

I used to achieve it by generating a client id in the plugin and using this id to recognize users in the server. Then tell the plugin the user information. I don't know how to implement this in supabase. Any ideas are welcomed. Thanks!

**Comment by izhan** [comment]
@leadream did you ever solve this? We have a similar issue.

**Reply by hallee9000** [reply, marked as answer]
The solution is tricky. Steps are here:
1. Generate a uuid in plugin and combine a sign in link like /signin/:uuid;
2. Users click this link to open a web page, sign in
4. Send user info and uuid to server (store in a database)
5. The plugin queries from database by this uuid continually, until get the user info

<img width="983" alt="image" src="https://github.com/supabase/supabase/assets/12868055/36f97c61-75a3-4462-a142-3f4656265d4b">

**Reply by hafisn07** [reply]
@leadream @izhan Hey any sample code you could give us. That would be help full for us. Couldn’t found much documentation regarding the authentication for figma plugin other than the official figma doc which seems little confusing.

**Reply by rnrnshn** [reply]
Hi, Have you managed to find a solution to this?
